app/controllers/concerns/api/taxonomy_scope.rb

```
@@ -7,8 +7,12 @@ module TaxonomyScope
     end
 
     def set_taxonomy_scope
-      Location.current ||= @location = Location.find_by_id(params[:location_id]) if SETTINGS[:locations_enabled]
-      Organization.current ||= @organization = Organization.find_by_id(params[:organization_id]) if SETTINGS[:organizations_enabled]
+      if SETTINGS[:locations_enabled] && !params[:location_id].nil?
+        Location.current ||= @location = Location.find_by_id!(params[:location_id])
```

@isratrade  as I wrote in previous comment, When running the tests: Location.find_by_id(nil) raises exception instead of returning nil. I failed to find the cause yet. Any idea why it may be?